_posts/kevin_hartmann/2017-07-31-Why-Siri-Cant-Code.md

```diff
@@ -0,0 +1,481 @@
+
+2) **Roll the dice**
+<image src="{{ site.urlimg }}/kevin_hartmann/dice.jpg" /> 
+ 
+Try to analyze any program you're given. If you can't, at some point, give up and stop trying.
+
+If you found an answer, great! 
+
+If we gave up, we don't know the answer, but we also don't know _why_ we don't know it. One of those things we can't always know for all programs is whether the reason we failed to find an answer because the question can't be answered at all, or whether there was an answer that we could have found, but failed to.
+
+### How do we get started?
+
+This semantic analysis stuff sounds interesting and all, but how do we actually **do** it? Don't I need a PhD just to get started with this whole thing? Isn't it a purely research thing at this point?
+
+The answers to those questions might have been "yes" a short time ago, but there's a new computing framework out there, designed to make semantic analysis a whole lot easier: in practice, not just in theory!
+
+## Introducing The 𝕂 Framework for Semantic Analysis
+<image src="{{ site.urlimg }}/kevin_hartmann/fireworks.jpg" /> 
+
+[𝕂](http://www.kframework.org/) is an exciting new framework for doing semantic analysis of programming languages; from the ground up.
+
+First, we need to define the both the syntax and the semantics of our programming language, using 𝕂. After that, the 𝕂 framework provides you with a whole suite of tools that are designed to help you work with, understand, and analyze programs written in that language! 
+
+What sort of tools do we get? Well, according to the 𝕂 FAQ, you get "a parser, an interpreter, a state-space explorer ... and even a deductive program verifier." In other words, you provide the language; and the 𝕂 framework gives you the tools to run your language, to analyze programs in it, and to debug it; automatically!
+
+The 𝕂 framework takes the "roll the dice" approach to semantic analysis; but it tries hard to succeed in all the cases where it **can** succeed. If the designers had instead opted for the more restricting "stack the deck" approach, you would be limited in what kind of semantics you could use 𝕂 to define.
+
+As it is, the sky seems to be the limit for what semantics you can define using 𝕂! Not only can 𝕂 be used to define programming languages, the designers have made it capable of formally defining the semantics of such things as formal logic systems, and type systems, as well. 𝕂 is not only designed to handle the realities of real world programming language designs, the designers explicitly took all of difficult to implement features of modern programming languages, bundled them all together into a language definition called _"Challenge"_; then proceeded to use 𝕂 semantics to define the _Challenge_ language as an example of what their system can do!
```

Comment:
> [...] such things as formal logic systems, and type systems, as well

Reads a bit off
Perhaps just "such things as formal logic systems and type systems"